titus-server-master/src/main/java/com/netflix/titus/master/mesos/DefaultV3TaskInfoFactory.java

```diff
@@ -206,6 +202,18 @@ public DefaultV3TaskInfoFactory(MasterConfiguration masterConfiguration,
         return containerInfoBuilder;
     }
 
+    private void setImage(ContainerInfo.Builder containerInfoBuilder, Image image) {
+        containerInfoBuilder.setImageName(image.getName());
+        String registryUrl = mesosConfiguration.getRegistryUrl();
+        if (!Strings.isNullOrEmpty(registryUrl)) {
```

Comment:
Are we supporting missing registry URLs only while we roman ride agent builds or are we expecting a long-term need for this?

Reply:
This logic has to exist in order to support backwards compatibility on the executor. If the contract to the executor changes, we can update this logic accordingly.